heron/common/src/cpp/network/baseconnection.cpp

```diff
@@ -50,6 +53,67 @@ BaseConnection::BaseConnection(ConnectionEndPoint* endpoint, ConnectionOptions*
   mState = INIT;
   mOnClose = NULL;
   bufferevent_options boptions = BEV_OPT_DEFER_CALLBACKS;
+
+  if (mOptions->ssloptions_.is_configured()) {
+    LOG(INFO) << "SSL is configured with certificate path"
+              << mOptions->ssloptions_.get_certificate_path() << " and private key path "
+              << mOptions->ssloptions_.get_private_key_path();
+    openSSLConnection(boptions);
+  } else {
+    openConnection(boptions);
+  }
+}
+
+void BaseConnection::openSSLConnection(bufferevent_options boptions) {
+  LOG(INFO) << "Starting connection with SSL cert path " <<
+    mOptions->ssloptions_.get_certificate_path() << " and key path " <<
+    mOptions->ssloptions_.get_private_key_path();
+
+  SSL_CTX *ctx;
+  if ((ctx = SSL_CTX_new(SSLv23_method())) == NULL) {
+    sp_int64 error = ERR_get_error();
+    LOG(FATAL) << "Error creating ssl context, error " << error << " "
+               << ERR_error_string(error, NULL);
+  }
+
+  SSL_CTX_set_timeout(ctx, 5);  // in seconds
+
+  int status = SSL_CTX_use_certificate_chain_file(
+      ctx, mOptions->ssloptions_.get_certificate_path().c_str());
+  if (1 != status) {
+    sp_int64 error = ERR_get_error();
+    LOG(FATAL) << "Error loading ssl cert status " << status << " error " << error << " "
+               << ERR_error_string(error, NULL);
+  }
+
+  if (1 != SSL_CTX_use_PrivateKey_file(ctx, mOptions->ssloptions_.get_private_key_path().c_str(),
```

Comment:
you never set status equal to the return value of this function; the log statement below is misleading. an alternate way to write this is:

```
if ((status = SSL_CTX_...(...)) != 1) {
...
}
```